### PR TITLE
Add memory metering for fixed-size numbers

### PR DESCRIPTION
**Pull request by SupunS**
Work towards https://github.com/dapperlabs/cadence-private-issues/issues/2

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/interpreter/value.go [hunk 1]
```diff
@@ -12772,7 +12835,12 @@ func ConvertFix64(value Value) Fix64Value {
...
-		return Fix64Value(value)
+		return NewFix64Value(
+			memoryGauge,
+			func() int64 {
+				return int64(value)
+			},
+		)
 
 	case BigNumberValue:
 		v := value.ToBigInt()
```
Comment (turbolent) [inline runtime/interpreter/value.go:12846]: Needs to be metered before this allocation occurs. See e.g. `ConvertInt64`

**Review by turbolent — COMMENTED** [review]
Nice!

File: runtime/interpreter/value.go (hunk 1 above)
Comment (SupunS) [reply runtime/interpreter/value.go]: ah yes, I had the entire function as the converter, but then refactored to make it a bit cleaner, and have missed these two in the process. Fixed them in: b669fa35e80b4803f1ad35551c79062d4b4a8d0b

**Review by robert-e-davidson3 — APPROVED** [review]

**Comment by dsainati1** [comment]
All the fixed-size numbers use the same amount of memory (hence the fixed-size), so I don't see the need to use constructor functions for them like we did with strings. We can just have the creation functions for these values meter this constant amount and then do their normal operation.

**Comment by dsainati1** [comment]
Yea I guess my concern is that it complicates the API unnecessarily and results in extra computation because of the deferred function calls, when I think there is no possible harm done by metering the allocation after it happens in this case, since it is a fixed small size.

**Comment by SupunS** [comment]
This is the pattern we have followed for all other values (e.g all number types, all simple types, etc)

**Review by turbolent — APPROVED** [review]

**Comment by turbolent** [comment]
@SupunS Looks good! Didn't realize there were follow-up commits, sorry about the delay

**Pull request merged**